src/core/game/PlayerImpl.ts

```
@@ -705,6 +705,13 @@ export class PlayerImpl implements Player {
     spawnTile: TileRef,
     unitSpecificInfos: UnitSpecificInfos = {},
   ): UnitImpl {
+    if (this.mg.config().isUnitDisabled(type)) {
+      console.warn(
```

let's throw an exception here. because when you call buildUnit, the client code expects a Unit.